fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1997,6 +1999,50 @@ public Range getUnbuiltRange() {
         return result;
     }
 
+    /**
+     * Loads the current state of the record store asynchronously.
+     * @param context the record context in which to retrieve the record store state
+     * @return a future that will contain the state of the record store located at the given subspace
+     */
+    @Nonnull
+    @API(API.Status.UNSTABLE)
+    CompletableFuture<MutableRecordStoreState> loadRecordStoreStateAsync(@Nonnull FDBRecordContext context) {
```

Comment:
Alternatively, should this call the deprecated static method to avoid code duplication? Then when the static method is actually removed, I think this will need to be moved into this method then, but until then, we don't need to maintain parallel code paths.

Reply:
good idea. done now.